### PR TITLE
GS/HW: Allow palette lookups from depth and deswizzle manual deswizzles

### LINKED ISSUE #3610: Vampire Panic - Graphical distorations with hardware renderer
***PCSX2 version:***
both stable 1.6.0 and the latest dev build 1.7.0

***PCSX2 options:***
default options with both hardware renderers and using the "safest" preset

- **Note**: Generally it's recommended to have settings as default as possible and use the ``Safest`` preset. If there's no issue with these settings but there is an issue with some other non-default option, then please state it explicitly.

***Plugins used:***
default plugins

- **Note**: In case the issue is a graphical problem with the GSdx plugin, please provide basic information like 
I have provided a GS dump of the issue 
[gsdx_20200819232917.gs.xz.zip](https://github.com/PCSX2/pcsx2/files/5099122/gsdx_20200819232917.gs.xz.zip)

***Description of the issue:***
distorations of the graphics.
Screenshot : 
gsdx_20200819232917
Also this is using the recently released english fan translation patch.
***How to reproduce the issue:***
Just play the game by using any of the hardware renderers.
The game works fine with the software renderer.

***Last known version to work:***
no idea but it happens in both 1.6.0 and latest dev build.

***PC specifications:***
CPU: Intel i7 8550U
GPU: Radeon 530.
(yeah its a laptop but it runs pretty well with it).

### LINKED ISSUE #7528: [BUG]: Glitch in Arc the Lad - Twilight of the Spirits
### Describe the Bug

There's a weird overlay glitch in the intro of Arc the Lad. In software rendering everything looks fine and I think the background gets blurred instead.

### Reproduction Steps

Watch the intro.

### Expected Behavior

Blurred background like in software renderer and no visual glitch.

### PCSX2 Revision

v1.7.3669

### Operating System

Linux (64bit) - Specify Distro Below

### If Linux - Specify Distro

Arch

### CPU

AMD Ryzen 5 2600X Six-Core Processor

### GPU

Advanced Micro Devices, Inc. [AMD/ATI] Navi 14 [Radeon RX 5500/5500M / Pro 5500M] (rev c5)

### GS Settings

Sharp bilinear filtering

### Emulation Settings

Defaults

### GS Window Screenshots

Arc the Lad - Twilight of the Spirits_SCUS-97231_20221204081629

### Logs & Dumps

[Arc the Lad - Twilight of the Spirits_SCUS-97231_20221204081629.gs.gz](https://github.com/PCSX2/pcsx2/files/10148051/Arc.the.Lad.-.Twilight.of.the.Spirits_SCUS-97231_20221204081629.gs.gz)

### PR DESCRIPTION
**Pull request by refractionpcsx2**
### Description of Changes
Allows palettes to be used in P8H lookups on depth textures.
Deswizzles draws which are manually deswizzling depth to be used on a colour target.

Also added mipmapping to Arc the Lad to help with the depth of field effect.

### Rationale behind Changes
We don't swizzle the depth like the GS does, this would be slow if we did, but gamedevs who tried to be clever and deswizzle it themselves kinda screwed us there, so this replaces the draw with a single sprite.

If the Z is different on each sprite there is no handling for this, but it will throw a log message.

### Suggested Testing Steps
Test Crash N Burn, Arc the Lad - End of Darkness/Generation and Vampire Panic.  Maybe just random spattering of games. Mercenaries 1 & 2 also showed very small differences but nothing scary, maybe check them.

Fixes #7528 Arc the Lad - Twilight of the Spirits
Fixes #3610 Vampire Panic

Arc the Lad - End of Darkness:
Master:
image
PR:
image

Arc the Lad - Twilight of the Spirits:
Master:
image
PR:
image

Crash N Burn:
Master:
image
PR:
image

Vampire Panic:
Master:
image
PR:
image

Vampire Panic (second comparison):
Master:
image
PR:
image

### CODE REVIEW

File: bin/resources/shaders/vulkan/tfx.glsl [hunk 1]
```diff
@@ -680,7 +680,9 @@ vec4 sample_depth(vec2 st, ivec2 pos)
 		t = fetch_c(uv) * 255.0f;
 	}
 	#endif
-
+	
+
```
Comment (lightningterror) [inline bin/resources/shaders/vulkan/tfx.glsl:684]: Maybe leave only 1 empty line?

**Review by lightningterror — COMMENTED** [review]
Need to bump shader cache too.

**Comment by refractionpcsx2** [comment]
> Need to bump shader cache too.

done

File: bin/resources/shaders/vulkan/tfx.glsl [hunk 2]
```diff
@@ -680,7 +680,7 @@ vec4 sample_depth(vec2 st, ivec2 pos)
 		t = fetch_c(uv) * 255.0f;
 	}
 	#endif
-
+
```
Comment (lightningterror) [inline bin/resources/shaders/vulkan/tfx.glsl:683]: Tab.

**Review by JordanTheToaster — APPROVED** [review]

**Pull request merged**